File: Backend/Models/CourseOfferingMeeting.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models;
public class CourseOfferingMeeting
{
    [Key]
    public int MeetingId { get; set; }

    [Required]
    public Day Day { get; set; }

    public TimeSpan? StartTime { get; set; }
    public TimeSpan? EndTime { get; set; }

    public string? Location { get; set; }

    [Required]
    public int OfferingId { get; set; }

    [Required]
    public MeetingType Type { get; set; }

    [ForeignKey(nameof(OfferingId))]
    public CourseOffering Offering { get; set; } = null!;
}

File: Frontend/src/types/courseOfferingMeeting.ts
import type { MeetingType } from './meetingType.ts';
import type { Day } from './day.ts';

export interface CourseOfferingMeeting {
  meetingId: number;
  day: Day;
  startTime: string | null;
  endTime: string | null;
  location: string | null;
  type: MeetingType;
}
